src/rpc/util.cpp

```diff
@@ -62,24 +62,20 @@ void RPCTypeCheckObj(const UniValue& o,
 {
     for (const auto& t : typesExpected) {
         const UniValue& v = find_value(o, t.first);
-        if (!fAllowNull && v.isNull())
-            throw JSONRPCError(RPC_TYPE_ERROR, strprintf("Missing %s", t.first));
-
+        if (!fAllowNull && v.isNull()) {
+            RPCTypeCheckArgument(v, t.second);
+        }
         if (!(t.second.typeAny || v.type() == t.second.type || (fAllowNull && v.isNull()))) {
-            std::string err = strprintf("Expected type %s for %s, got %s",
```

Comment:
Only this string needs to be adjusted, per the created issue. Other strings should stay untouched.

Reply:
Sorry if this was unclear, but only the *type* error string should be adjusted, not the error string for missing fields or unexpected fields, or other error strings.